### PR TITLE
fix: Skip static calls which are made using the class name

### LINKED ISSUE #8688: LocallyCalledStaticMethodToNonStaticRector does not skip when doing static call using class name
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | last dev-main             |

## Minimal PHP Code Causing Issue
See https://getrector.com/demo/81b063a7-d8ae-430b-be7d-172d411f5b7d
```
<?php

declare(strict_types=1);

namespace Rector\Tests\CodeQuality\Rector\ClassMethod\LocallyCalledStaticMethodToNonStaticRector\Fixture;

final class SkipPrivateStaticCallFromStatic
{
    public static function bar(string $input)
    {
        return SkipPrivateStaticCallFromStatic::reallyBar($input);
    }

    private static function reallyBar($input)
    {
        return $input . ' - bar';
    }
}
````

## Expected Behaviour
Rector should skip if static call is done using the class name instead of self / static

### PR DESCRIPTION
**Pull request by nikolicaleksa**
Fixes https://github.com/rectorphp/rector/issues/8688

### CODE REVIEW

File: rules-tests/CodeQuality/Rector/ClassMethod/LocallyCalledStaticMethodToNonStaticRector/Fixture/skip_static_call_from_static.php.inc [hunk 1]
```diff
@@ -11,8 +11,28 @@ final class SkipPrivateStaticCallFromStatic
...
+
+    private static function reallyFoo($input)
+    {
+        return $input . ' - foo';
+    }
+
+    private static function reallyBar($input)
+    {
+        return $input . ' - bar';
+    }
```
Comment (TomasVotruba) [inline rules-tests/CodeQuality/Rector/ClassMethod/LocallyCalledStaticMethodToNonStaticRector/Fixture/skip_static_call_from_static.php.inc]: Could you extract this one into a new test fixture file? Just so we have git history fixture tight with a specific fix

Comment (nikolicaleksa) [reply rules-tests/CodeQuality/Rector/ClassMethod/LocallyCalledStaticMethodToNonStaticRector/Fixture/skip_static_call_from_static.php.inc]: Done, I've left the test case for self as I did not change anything for it, it was just missing

File: rules/CodeQuality/Rector/ClassMethod/LocallyCalledStaticMethodToNonStaticRector.php [hunk 2]
```diff
@@ -151,6 +151,7 @@ private function refactorClassMethod(Class_ $class, ClassMethod $classMethod): ?
      */
     private function isClassMethodCalledInAnotherStaticClassMethod(Class_ $class, ClassMethod $classMethod): bool
     {
+        $currentClassNamespacedName = implode('\\', $class->namespacedName?->getParts() ?? []);
```
Comment (samsonasik) [inline rules/CodeQuality/Rector/ClassMethod/LocallyCalledStaticMethodToNonStaticRector.php]: is this can be use `(string) $this->getName($class);` ?

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Let's give it a try, thank you @nikolicaleksa

**Pull request merged**